src/main/webapp/script.js

```diff
@@ -30,41 +27,133 @@ function passQuery() {
   const locationInput = query.get('location');
 
   const personType = document.querySelector(
-      '#person-type option[value=\'' + personTypeInput + '\']').dataset.value;
+    '#person-type option[value=\'' + personTypeInput + '\']').dataset.value;
   const action = document.querySelector(
       '#action option[value=\'' + actionInput + '\']').dataset.value;
   const location = document.querySelector(
-      '#location option[value=\'' + locationInput + '\']').dataset.value;
+    '#location option[value=\'' + locationInput + '\']').dataset.value;
 
   const actionToText = new Map();
   actionToText['moved'] = 'moved to';
   if (actionToText.has(action)) {
     action = actionToText[action];
   }
   const description = {action: action, age: personType};
-  const region =
-      location === 'state' ? 'U.S. state' : 'State Name county';
+  const isCountyQuery = location !== 'state';
+  const region = isCountyQuery ? locationInput + ' county' : 'U.S. state';
   const title = 'Population who ' + action +
       ' in each ' + region + ' (' + personType.replace('-', ' ') + ')';
+  document.getElementById('map-title').innerText = title;
 
   const fetchUrl = '/query?person-type=' + personType +
-      '&action=' + action +
-      '&location=' + location;
+    '&action=' + action +
+    '&location=' + location;
+
   fetch(fetchUrl)
     .then((response) => {
       if (response.ok) {
         response.json().then((jsonResponse) => JSON.parse(jsonResponse))
-        .then((data) => {
+        .then((censusDataArray) => {
           // data is a 2D array, where the first row is a header row and all
           // subsequent rows are one piece of data (e.g. for a state or county)
-          displayVisualization(data, description, title);
+          const data = createDataArray(censusDataArray, isCountyQuery);
+          displayAmChartsMap(data, description, location, isCountyQuery);
+          document.getElementById('more-info').innerText = '';
         });
       } else {
         displayError(response.status, response.statusText);
       }
     });
 }
 
+function displayAmChartsMap(data, description, location, isCountyQuery) {
+  am4core.useTheme(am4themes_animated);
+  const chart = am4core.create('map', am4maps.MapChart);
+  chart.height = 550;
+  chart.zoomControl = new am4maps.ZoomControl();
+  // only allow zooming with buttons
+  chart.mouseWheelBehavior = 'none';
+
+  // Create map instance
+  if (isCountyQuery) {
+    chart.geodata = location ===
+        '06' ? am4geodata_region_usa_caLow : am4geodata_region_usa_njLow;
```

Comment:
for now this is ok but as we add more states we should look to have a clean way to store this mapping. For example this could also be part of the stateinfo.